cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -756,6 +757,8 @@ pub mod pallet {
 	pub(super) type DidSetValidationCode<T: Config> = StorageValue<_, bool, ValueQuery>;
 
 	/// The relay chain block number associated with the last parachain block.
+	///
+	/// This is updated in `on_finalize`.
```

Comment:
It kind of begs the question; how was it updated before this PR?

Reply:
Did you looked at the code this pr is modifying? This answers your question.

Reply:
I can see how it works _after_ the PR changes, not _before_. But fine, I'll look for it myself.

Reply:
It was updated at the end of the inherent.